linalg/hypre.cpp

```diff
@@ -1858,46 +2484,69 @@ HypreParMatrix * ParMult(const HypreParMatrix *A, const HypreParMatrix *B,
 
 HypreParMatrix * RAP(const HypreParMatrix *A, const HypreParMatrix *P)
 {
+   hypre_ParCSRMatrix * rap;
+
+#ifdef HYPRE_USING_CUDA
+   // FIXME: this way of computing Pt A P can completely eliminate zero rows
+   //        from the sparsity pattern of the product which prevents
+   //        EliminateZeroRows() from working correctly. This issue is observed
+   //        in ex28p.
+   // Quick fix: add a diagonal matrix with 0 diagonal.
+   // Maybe use hypre_CSRMatrixCheckDiagFirst to see if we need the fix.
```

Comment:
We decided not to pursue these fixes in mfem-4.3, correct?

Reply:
Yes, we can postpone this.

Reply:
ok